### PR TITLE
Pypi packaging

### PR DESCRIPTION
**Pull request by bpkroth**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -74,6 +74,31 @@ We expect development to follow a typical "forking" style workflow:
 4. PRs are associated with [Github Issues](https://github.com/microsoft/MLOS/issues) and need [MLOS-committers](https://github.com/orgs/microsoft/teams/MLOS-committers) to sign-off (in addition to other CI pipeline checks like tests and lint checks to pass).
 5. Once approved, the PR can be completed using a squash merge in order to keep a nice linear history.
 
+## Distributing
```
Comment (motus) [inline CONTRIBUTING.md:77]:
```suggestion
### Distributing
```
or did you mean it?

File: Makefile [hunk 2]
```diff
@@ -232,6 +232,8 @@ mlos_bench/dist/tmp/mlos-bench-latest.tar: PACKAGE_NAME := mlos-bench
 	# Check to make sure the mlos_bench module has the config directory.
 	[ "$(MODULE_NAME)" != "mlos_bench" ] || unzip -t $(MODULE_NAME)/dist/$(MODULE_NAME)-*-py3-none-any.whl | grep -m1 mlos_bench/config/
 	cd $(MODULE_NAME)/dist/tmp && ln -s ../$(MODULE_NAME)-*-py3-none-any.whl $(MODULE_NAME)-latest-py3-none-any.whl
+	# Check to make sure the README contents made it into the package metadata.
+	unzip -p $(MODULE_NAME)/dist/tmp/$(MODULE_NAME)-latest-py3-none-any.whl */METADATA | egrep -v '^[A-Z][a-zA-z-]+:' | grep -q -i '^# mlos'
```
Comment (motus) [inline Makefile:236]:
```suggestion
	unzip -p $(MODULE_NAME)/dist/tmp/$(MODULE_NAME)-latest-py3-none-any.whl */METADATA | egrep -v '^[A-Z][a-zA-Z-]+:' | grep -q -i '^# mlos'
```
a typo?

File: mlos_bench/setup.py [hunk 3]
```diff
@@ -6,14 +6,46 @@
...
+# temp directory.
+# Similarly, we can't use a utility script outside this module, so this code has to
+# be duplicated for now.
+def _get_long_desc_from_readme(base_url: str) -> dict:
+    pkg_dir = os.path.dirname(__file__)
+    readme_path = os.path.join(pkg_dir, 'README.md')
+    if not os.path.isfile(readme_path):
+        return {}
+    jsonc_re = re.compile(r'```jsonc')
+    link_re = re.compile(r'\]\(([^:#)]+)(#[a-zA-Z0-9_-]+)?\)')
```
Comment (motus) [inline mlos_bench/setup.py:36]: shorter
```suggestion
    link_re = re.compile(r'\]\(([^:#)]+)(#[\w-]+)?\)')
```

File: mlos_core/setup.py [hunk 4]
```diff
@@ -25,6 +30,34 @@
...
+# be duplicated for now.
+# Also, to avoid caching issues when calculating dependencies for the devcontainer,
+# we return nothing when the file is not available.
+def _get_long_desc_from_readme(base_url: str) -> dict:
+    pkg_dir = os.path.dirname(__file__)
+    readme_path = os.path.join(pkg_dir, 'README.md')
+    if not os.path.isfile(readme_path):
+        return {}
+    jsonc_re = re.compile(r'```jsonc')
+    link_re = re.compile(r'\]\(([^:#)]+)(#[a-zA-Z0-9_-]+)?\)')
```
Comment (motus) [inline mlos_core/setup.py:48]:
```suggestion
    link_re = re.compile(r'\]\(([^:#)]+)(#[\w-]+)?\)')
```

**Review by motus — APPROVED** [review]
Looks good! left a few comments - all minor, except for one potential typo in regex

File: Makefile (hunk 2 above)
Comment (motus) [reply Makefile:236]: or
```suggestion
	unzip -p $(MODULE_NAME)/dist/tmp/$(MODULE_NAME)-latest-py3-none-any.whl */METADATA | egrep -v '^[A-Z][\w-]+:' | grep -q -i '^# mlos'
```

File: mlos_core/setup.py (hunk 4 above)
Comment (bpkroth) [reply mlos_core/setup.py:48]: This is the same regex used in copy-source-tree-docs.sh and I know this one works as is right now.  Retesting changes for that would be a little bit of a chore, so let's fix this later on if we run into another issue that needs it.

File: CONTRIBUTING.md (hunk 1 above)
Comment (bpkroth) [reply CONTRIBUTING.md:77]: No, I think those are slightly separate topics.

**Pull request merged**